packages/beacon-node/src/api/impl/beacon/state/utils.ts

```diff
@@ -254,7 +253,6 @@ async function getFinalizedState(
     state = stateTransition(state, block, {
       // Replaying finalized blocks, all data is considered valid
       executionPayloadStatus: ExecutionPayloadStatus.valid,
-      dataAvailableStatus: DataAvailableStatus.available,
```

Comment:
The main purpose of this notation was to force all consumers to explicitly declare assumptions about blobs. We will have to support different paths like regen block replay where we don't want to load blobs, but already know they are available. I'm aware it's technically pointless but being verbose about such critical decisions sounded good to me. I'm fine moving DA check outside of state transition, but this explicit statements should stay IMO

Reply:
ok sounds good, will make modifications